src/ServicePulse.Host/vue/src/stores/HeartbeatsStore.ts

```diff
@@ -0,0 +1,137 @@
+import { usePatchToServiceControl, useTypedFetchFromServiceControl } from "@/composables/serviceServiceControlUrls";
+import { acceptHMRUpdate, defineStore } from "pinia";
+import { computed, ref } from "vue";
+import useAutoRefresh from "@/composables/autoRefresh";
+import { Endpoint, EndpointStatus } from "@/resources/Heartbeat";
+import moment from "moment";
+import SortOptions, { SortDirection } from "@/resources/SortOptions";
+import { getSortFunction } from "@/components/OrderBy.vue";
+import { useCookies } from "vue3-cookies";
+
+export enum DisplayType {
+  Instances = "Endpoint Instances",
+  Logical = "Logical Endpoints",
+}
+
+function mapEndpointsToLogical(endpoints: Endpoint[]) {
+  const logicalNames = [...new Set(endpoints.map((endpoint) => endpoint.name))];
+  return logicalNames.map((endpointName) => {
+    const logicalList = endpoints.filter((endpoint) => endpoint.name === endpointName);
+    const aliveList = logicalList.filter((endpoint) => endpoint.monitor_heartbeat && endpoint.heartbeat_information && endpoint.heartbeat_information.reported_status === EndpointStatus.Alive);
+
+    var aliveCount = aliveList.length;
+    var downCount = logicalList.filter((endpoint) => endpoint.monitor_heartbeat).length - aliveCount;
+
+    return {
+      name: endpointName,
+      aliveCount: aliveCount,
+      downCount: downCount,
+      heartbeat_information: {
+        reported_status: aliveCount > 0 ? EndpointStatus.Alive : EndpointStatus.Dead,
+        last_report_at: logicalList.reduce((previousMax: Endpoint | null, endpoint: Endpoint) => {
+          if (endpoint.heartbeat_information) {
+            if (previousMax) {
+              return moment.utc(endpoint.heartbeat_information.last_report_at) > moment.utc(previousMax.heartbeat_information!.last_report_at) ? endpoint : previousMax;
+            }
+            return endpoint;
+          }
+          return previousMax;
+        }, null)?.heartbeat_information?.last_report_at,
+      },
+      monitor_heartbeat: logicalList.some((endpoint) => endpoint.monitor_heartbeat),
+    } as Endpoint;
+  });
+}
+
+export const sortOptions: SortOptions<Endpoint>[] = [
+  {
+    description: "Name",
+    selector: (group) => group.name,
+    icon: "bi-sort-alpha-",
+  },
+  {
+    description: "Latest heartbeat",
+    selector: (group) => group.heartbeat_information?.last_report_at ?? "",
+    icon: "bi-sort-",
+  },
+];
+
+export const useHeartbeatsStore = defineStore("HeartbeatsStore", () => {
+  const cookies = useCookies().cookies;
+
+  const selectedDisplay = ref(cookies.get("heartbeats_display_type") ?? DisplayType.Instances);
```

Comment:
Could be a computed wrapped around cookies

Reply:
doesn't work - the getter loses its reactivity somehow